src/app/examples/crud/components/crud.component.ts

```diff
@@ -40,13 +47,20 @@ export class CrudComponent implements OnInit, OnDestroy {
     };
   }
 
-  constructor(public store: Store<State>, public fb: FormBuilder) {}
+  constructor(
+    public store: Store<State>,
+    public fb: FormBuilder,
+    private cd: ChangeDetectorRef
+  ) {}
 
   ngOnInit() {
     this.books$ = this.store.pipe(select(selectAllBooks));
     this.store
       .pipe(select(selectSelectedBook), takeUntil(this.unsubscribe$))
-      .subscribe(book => (this.selectedBook = book));
+      .subscribe(book => {
+        this.selectedBook = book;
+        this.cd.markForCheck();
```

Comment:
Lets rework this component to **NOT** subscribe to this in the `ngOnInit` but in the template using `*ngIf=selectedBook$ | async as selectedBook` so that we don't have to manually trigger `markForCheck`. Let's do this for all cases when this is needed.

Another side effec could be that many of these components will no longer need `OnDestroy` and  `private unsubscribe$: Subject<void> = new Subject<void>();`...

Reply:
Deal 👍  

I didn't want to refactor a lot without approval, but you are right in the main comment, this is the point. With that:

1. we'll have a more declarative syntax,
2. remove manual unsubscriptions (automatically we remove possibility for human error having a memory leak if forgot to unsubscribe) 
3. make in general components easier to reason about because we don't need to keep local state (This also comes with the benefit that it prevents developers accidentally mutating the data got from the store, since they will not operate on it, components just get what they need and pass it down).

Reply:
@simply10w cool, the points make sense. I would like to add that there is one disadvantage and that is that some of the **local** state is now kept in the template, but I have to agree it's usualy pretty minimal. If we follow naming convention like `something$` is then unwrapped in template to `something` it should be easy to navigate and reason about.

Anyway looking forward to the final PR, thank you again!

Reply:
@simply10w  I am in favor of the second example where we pass `bookId` from the template.

Reply:
@simply10w btw thanks for a very nice communication of the proposals, very appreciated!

Reply:
@tomastrajan You're welcome, i'm really glad to be a part of this and help out.

Reply:
Agreeing with everything that is said here.
The only thing I want to add it, if we find that the HTML will become to bloated/big we can always split it up into multiple components.

Reply:
Splitting up might be a good idea even for performance reasons, for example in `todos-container` the todo list is in the same template with the `big-input`, every event that happens on big-input will trigger change detection over the entire list and each item has a lot of bindings(mostly translation pipes).

But if we encapsulate the list in a component with OnPush, if input has not changed it won't need to check the list at all.

Reply:
@simply10w yeah that's true, there is a lot of room for breaking stuff up into multiple components. I think that's also one of the killer features of Angular that it is possible to come up with working solution in one "big template" which then can be broken up further into more discrete components.

Reply:
@tomastrajan Yea, it gives you a lot of flexibility to quickly prototype and later when you can more clearly see correct abstractions, pull them out with ease.